proposed/0007-Address_relocation.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: Prevent any single address being used more than once during address relocation.
+- Type enhancement
+- Related components routing
+- Start Date: 21-09-2015
+- RFC PR: (leave this empty)
+- Issue number: (leave this empty)
+
+# Summary
+
+Prevent a potential attack where a node gains overall/majority influence in a group setting.
+
+# Motivation
+
+The possibility 'may' exist for a node on the network to generate off-line derived identities close to a specific target and to then connect nodes whose resultant network relocated identities actually lie close to that target. In such cases the owner of that node will dominate, or at least have a dominant influence within, the group close to that target. This RFC proposes an amendment to the current process to prevent this possibility arising in practice.
+
```

Comment:
I think the main purpose of using address-relocation is to avoid `user defined identity` to be used as vault's address.
The complexity of attack under `K closer to T` and `Hash(K + N1 + N2) closer to T` is the same and will be easy to achieve, as long as user provided identify can be used (i.e. attack modify the routing's ID generating algorithm to using fixed K).
Such attack can only be avoided only when a joining node's address is granted by the network dynamically.